From aaf496be3e1cf567ac4d7b9ab927a2ba7cb486d2 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Wed, 23 Jul 2025 15:43:33 -0300
Subject: [PATCH] Fix declaration of localeconv()

---
 libstdc++-v3/include/c_global/clocale | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libstdc++-v3/include/c_global/clocale b/libstdc++-v3/include/c_global/clocale
index fd30f1707..d481ad2e8 100644
--- a/libstdc++-v3/include/c_global/clocale
+++ b/libstdc++-v3/include/c_global/clocale
@@ -54,7 +54,9 @@ namespace std
 {
   using ::lconv;
   using ::setlocale;
+#if defined(__ANDROID__) && __ANDROID_API__ >= 21
   using ::localeconv;
+#endif
 } // namespace std
 
 #endif
-- 
2.49.0

