From bbf0b5107a4ef1c78140d7daa387baa1c2311f85 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 3 Jun 2025 17:46:19 -0300
Subject: [PATCH] a

---
 libstdc++-v3/include/c_compatibility/stdio.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libstdc++-v3/include/c_compatibility/stdio.h b/libstdc++-v3/include/c_compatibility/stdio.h
index 7241481..b9e90b3 100644
--- a/libstdc++-v3/include/c_compatibility/stdio.h
+++ b/libstdc++-v3/include/c_compatibility/stdio.h
@@ -72,7 +72,9 @@ using std::puts;
 using std::ungetc;
 using std::fread;
 using std::fwrite;
+#if defined(__ANDROID__) && __ANDROID_API__ >= 24
 using std::fgetpos;
+#endif
 using std::fseek;
 using std::fsetpos;
 using std::ftell;
-- 
2.49.0

