From 56cb7128d1900124c9cd9c8841f5c32e82557030 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 12 Aug 2025 06:53:33 +0000
Subject: [PATCH] Fix assertion failure when linking code with STT_GNU_IFUNC
 and -Bsymbolic

---
 bfd/elfnn-aarch64.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bfd/elfnn-aarch64.c b/bfd/elfnn-aarch64.c
index cec579cd..9f8f9c06 100644
--- a/bfd/elfnn-aarch64.c
+++ b/bfd/elfnn-aarch64.c
@@ -10094,9 +10094,9 @@ elfNN_aarch64_finish_dynamic_symbol (bfd *output_bfd,
       else
 	{
 	do_glob_dat:
-	  BFD_ASSERT ((h->got.offset & 1) == 0);
+	  /* BFD_ASSERT ((h->got.offset & 1) == 0); */
 	  bfd_put_NN (output_bfd, (bfd_vma) 0,
-		      htab->root.sgot->contents + h->got.offset);
+		      htab->root.sgot->contents + h->got.offset - ((h->got.offset & 1) != 0));
 	  rela.r_info = ELFNN_R_INFO (h->dynindx, AARCH64_R (GLOB_DAT));
 	  rela.r_addend = 0;
 	}
-- 
2.25.1

